# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: layout/base/nsLayoutUtils.h
# Commit: 8079db41c325
# Full Hash: 8079db41c32526a9535056ce616c5383dc2a56c1
# Author: Timothy Nikkel <tnikkel@gmail.com>
# Date: 2022-08-29 21:45:26
# Description:
#   Bug 1787684. Only return views with widgets from FindFloatingViewContaining. r=emilio
#   
#   The code that bug 1754436 deleted checked to make sure there was a widget for the returned view, but it looks like it's possible in some edge cases to have an open popup that doesn't have a widget (most likely just for a split second). We need a widget to dispatch the event, and if it doesn't have a widget it's not visible so no point in touching it at all anyways.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D155819
# ==============================================================================

diff -r 251d145c0b4e -r 8079db41c325 layout/base/nsLayoutUtils.h
--- a/layout/base/nsLayoutUtils.h	Mon Aug 29 09:46:13 2022 +0000
+++ b/layout/base/nsLayoutUtils.h	Mon Aug 29 09:58:11 2022 +0000
@@ -754,9 +754,13 @@
    * @return        Null, if there is no popup frame at the point, otherwise,
    *                returns top-most popup frame at the point.
    */
+  enum class GetPopupFrameForPointFlags : uint8_t {
+    OnlyReturnFramesWithWidgets = 0x1,
+  };
   static nsIFrame* GetPopupFrameForPoint(
       nsPresContext* aRootPresContext, nsIWidget* aWidget,
-      const mozilla::LayoutDeviceIntPoint& aPoint);
+      const mozilla::LayoutDeviceIntPoint& aPoint,
+      GetPopupFrameForPointFlags aFlags = GetPopupFrameForPointFlags(0));
 
   /**
    * Get container and offset if aEvent collapses Selection.
@@ -3052,6 +3056,7 @@
 };
 
 MOZ_MAKE_ENUM_CLASS_BITWISE_OPERATORS(nsLayoutUtils::PaintFrameFlags)
+MOZ_MAKE_ENUM_CLASS_BITWISE_OPERATORS(nsLayoutUtils::GetPopupFrameForPointFlags)
 
 template <typename PointType, typename RectType, typename CoordType>
 /* static */ bool nsLayoutUtils::PointIsCloserToRect(
